### PR TITLE
feat: add bepinex into vrising egg

### PR DESCRIPTION
**Pull request by Upd4ting**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Did you branch your changes and PR from that branch and not from your master branch?

### Changes to an existing Egg:

1. [x] Have you added an explanation of what your changes do and why you'd like us to include them?
2. [x] Have you tested your Egg changes?

### CODE REVIEW

**Comment by lilkingjr1** [comment]
This should be it's own egg, especially if it requires manual adjustment of the startup command. Please see my concerns/remarks in issue #1706.

**Comment by Upd4ting** [comment]
@lilkingjr1 I reverted the change on the vrising egg and created a new one.

**Comment by lilkingjr1** [comment]
@Upd4ting Thank you for doing this. Also, please update the README of the new egg to have any relevant information you think it may need. At the very least, change the name on the README and the name of the egg file to differentiate. You can leave me as an author on README if you want, but please remove me as a "Maintainer".

**Comment by Upd4ting** [comment]
@lilkingjr1 No problem! The readme has been updated :)

**Comment by lilkingjr1** [comment]
@Upd4ting Thank you. Please also rename the egg file to `egg-v-rising-bepinex.json` if you could. Lastly, I think there are some things with your egg that could be improved (ie. putting the needed `WINDLLOVERRIDES` code in the startup command), but I will leave this to you and anyone else interested in bepinex to sort out.

**Comment by Upd4ting** [comment]
@lilkingjr1 Thanks to you! My bad I though I had changed all of that in the egg. It should be done now.

**Comment by TwoReaperz** [comment]
I have a issue after install. The Server not starting. 

> ========== END OF STACKTRACE ===========

A crash has been intercepted by the crash handler. For call stack and other deta
ils, see the latest crash report generated in:
 * C:/users/container/Temp/Stunlock Studios/VRisingServer/Crashes
01b8:fixme:imm:ImeSetActiveContext (0x162a30, 1): stub
01b8:fixme:imm:ImmReleaseContext (00000000000100B4, 0000000000162A30): stub
wine: Unhandled exception 0x80000003 in thread 24 at address 0000000180EE7265 (thread 0024), starting debugger...
01c8:fixme:imm:ImeSetActiveContext (0x34d90, 1): stub
01c8:fixme:imm:ImmReleaseContext (00000000000100DE, 0000000000034D90): stub
01b8:fixme:event:wait_for_withdrawn_state window 0x100a8/1c00001 wait timed out
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub

**Comment by gOOvER** [comment]
i think, this BepInex Versiuon needs a higher Wine Version

I found this in Code:

`Your wine version doesn't support CoreCLR properly, expect crashes! Upgrade to wine 7.16 or higher`

File: game_eggs/steamcmd_servers/v_rising_bepinex/egg-v-rising-bepinex.json [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+  "exported_at": "2022-08-12T23:45:18+02:00",
+  "name": "V Rising BepInEx",
+  "author": "rehlmgaming@gmail.com",
+  "description": "Awaken as a vampire. Hunt for blood in nearby settlements to regain your strength and evade the scorching sun to survive. Raise your castle and thrive in an ever-changing open world full of mystery. Gain allies online and conquer the land of the living.",
+  "features": ["steam_disk_space"],
+  "docker_images": {
+    "ghcr.io/parkervcp/yolks:wine_latest": "ghcr.io/parkervcp/yolks:wine_latest"
+  },
+  "file_denylist": [],
+  "startup": "WINEDLLOVERRIDES=\"winhttp=n,b\" wine64 ./VRisingServer.exe -persistentDataPath save-data -address 0.0.0.0",
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/v_rising_bepinex/egg-v-rising-bepinex.json:16]: `export WINEDLLOVERRIDES="mscoree=n,b";` is the correct way for startup

**Comment by gOOvER** [comment]
the base libs are not download:

```
[Message: Preloader] BepInEx 6.0.0-pre.1 - VRisingServer
[Info   :   BepInEx] System platform: Windows 7 64-bit
[Info   :   BepInEx] Process bitness: 64-bit (x64)
[Info   : Preloader] Running under Unity v2020.3.31
[Message: Preloader] Downloading unity base libraries
[Fatal  :Unhollower] Failed to generate unhollowed assemblies: System.Net.Http.H
ttpRequestException: An error occurred while sending the request ---> System.Net
.WebException: Error: NameResolutionFailure
  at System.Net.WebConnection.Connect (System.Net.WebOperation operation, System
.Threading.CancellationToken cancellationToken) [0x00044] in <0cb84a912f9e474b97
78c5c9aac20660>:0 
  at System.Net.WebConnection.InitConnection (System.Net.WebOperation operation,
 System.Threading.CancellationToken cancellationToken) [0x000cc] in <0cb84a912f9
e474b9778c5c9aac20660>:0 
  at System.Net.WebOperation.Run () [0x0009a] in <0cb84a912f9e474b9778c5c9aac206
60>:0 
  at System.Net.WebCompletionSource`1[T].WaitForCompletion () [0x00094] in <0cb8
4a912f9e474b9778c5c9aac20660>:0 
  at System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.T
ask`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Fun
c`1[TResult] aborted, System.Threading.CancellationTokenSource cts) [0x000f8] in
 <0cb84a912f9e474b9778c5c9aac20660>:0
  at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult)
[0x00020] in <0cb84a912f9e474b9778c5c9aac20660>:0 
  at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (System.IA
syncResult iar, System.Func`2[T,TResult] endFunction, System.Action`1[T] endActi
on, System.Threading.Tasks.Task`1[TResult] promise, System.Boolean requiresSynch
ronization) [0x0000f] in <986ed57b9a8f4699a3c59a69eb05944a>:0 
--- End of stack trace from previous location where exception was thrown ---

  at System.Net.Http.MonoWebRequestHandler.SendAsync (System.Net.Http.HttpReques
tMessage request, System.Threading.CancellationToken cancellationToken) [0x0041b
] in <4535ed4534f04d0994ac41b63161cc92>:0 
   --- End of inner exception stack trace ---
  at System.Net.Http.MonoWebRequestHandler.SendAsync (System.Net.Http.HttpReques
tMessage request, System.Threading.CancellationToken cancellationToken) [0x00476
] in <4535ed4534f04d0994ac41b63161cc92>:0 
  at System.Net.Http.HttpClient.SendAsyncWorker (System.Net.Http.HttpRequestMess
age request, System.Net.Http.HttpCompletionOption completionOption, System.Threa
ding.CancellationToken cancellationToken) [0x000e8] in <4535ed4534f04d0994ac41b6
3161cc92>:0 
  at System.Net.Http.HttpClient.GetStreamAsync (System.String requestUri) [0x000
7d] in <4535ed4534f04d0994ac41b63161cc92>:0 
  at BepInEx.IL2CPP.ProxyAssemblyGenerator+AppDomainRunner.GenerateAssembliesInt
ernal (BepInEx.IL2CPP.ProxyAssemblyGenerator+AppDomainListener listener, System.
String unityVersion, BepInEx.IL2CPP.ProxyAssemblyGenerator+IL2CPPDumperType dump
erType) [0x00080] in <e9997477cd8143c9a348224def0a337e>:0 
  at BepInEx.IL2CPP.ProxyAssemblyGenerator+AppDomainRunner.GenerateAssemblies (B
epInEx.IL2CPP.ProxyAssemblyGenerator+AppDomainListener listener, System.String u
nityVersion, BepInEx.IL2CPP.ProxyAssemblyGenerator+IL2CPPDumperType dumperType)
[0x00000] in <e9997477cd8143c9a348224def0a337e>:0 
[Fatal  : Preloader] System.TypeInitializationException: The type initializer fo
r 'UnhollowerBaseLib.Runtime.UnityVersionHandler' threw an exception. ---> Syste
m.TypeLoadException: Could not load type of field 'UnhollowerRuntimeLib.Delegate
Support+<>c:<>9__7_0' (2) due to: Could not load file or assembly 'Il2Cppmscorli
b, Version=3.7.1.6, Culture=neutral, PublicKeyToken=null' or one of its dependen
cies.
  at (wrapper managed-to-native) System.RuntimeTypeHandle.type_is_assignable_fro
m(System.Type,System.Type)
  at System.RuntimeTypeHandle.CanCastTo (System.RuntimeType type, System.Runtime
Type target) [0x00000] in <986ed57b9a8f4699a3c59a69eb05944a>:0 
  at System.RuntimeType.IsAssignableFrom (System.Type c) [0x00020] in <986ed57b9
a8f4699a3c59a69eb05944a>:0 
  at UnhollowerBaseLib.Runtime.UnityVersionHandler+<>c__DisplayClass19_1.<.cctor
>b__2 (System.Type i) [0x00000] in <e00b54b0593445e4abdce314024df595>:0 
  at System.Linq.Enumerable.Any[TSource] (System.Collections.Generic.IEnumerable
`1[T] source, System.Func`2[T,TResult] predicate) [0x0002c] in <c8122f496f05432b
aca55ffb7d139a58>:0 
  at UnhollowerBaseLib.Runtime.UnityVersionHandler+<>c__DisplayClass19_0.<.cctor
>b__1 (System.Type t) [0x0001a] in <e00b54b0593445e4abdce314024df595>:0 
  at System.Linq.Enumerable+WhereArrayIterator`1[TSource].MoveNext () [0x0002c]
in <c8122f496f05432baca55ffb7d139a58>:0 
  at UnhollowerBaseLib.Runtime.UnityVersionHandler..cctor () [0x0014c] in <e00b5
4b0593445e4abdce314024df595>:0 
   --- End of inner exception stack trace ---
  at BepInEx.IL2CPP.Preloader.Run () [0x0017e] in <e9997477cd8143c9a348224def0a3
37e>:0
```

**Comment by gOOvER** [comment]
> I have a issue after install. The Server not starting.
> 
> Step: Downloading Egg Change port install egg Create Server
> 
> > ========== END OF STACKTRACE ===========
> 
> A crash has been intercepted by the crash handler. For call stack and other deta ils, see the latest crash report generated in:
> 
> * C:/users/container/Temp/Stunlock Studios/VRisingServer/Crashes
>   01b8:fixme:imm:ImeSetActiveContext (0x162a30, 1): stub
>   01b8:fixme:imm:ImmReleaseContext (00000000000100B4, 0000000000162A30): stub
>   wine: Unhandled exception 0x80000003 in thread 24 at address 0000000180EE7265 (thread 0024), starting debugger...
>   01c8:fixme:imm:ImeSetActiveContext (0x34d90, 1): stub
>   01c8:fixme:imm:ImmReleaseContext (00000000000100DE, 0000000000034D90): stub
>   01b8:fixme:event:wait_for_withdrawn_state window 0x100a8/1c00001 wait timed out
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub
>   00a0:fixme:file:CancelSynchronousIo (000000000000007C): stub

Fixme's are no crash ;)

**Comment by QuintenQVD0** [comment]
A offical wine staging images is releases: `ghcr.io/parkervcp/yolks:wine_staging`

**Comment by gOOvER** [comment]
can be closed. Completed with #2067

**Comment by QuintenQVD0** [comment]
Closed as of #2067

**Pull request closed without merging**